### PR TITLE
gopts: directly override config

### PR DESCRIPTION
**Pull request by kodebach**
Implements #3607

## Basics

These points need to be fulfilled for every PR:

- [ ] Short descriptions of your changes are in the release notes
      (added as entry in `doc/news/_preparation_next_release.md` which
      contains `_(my name)_`)
      **Please always add something to the release notes.**
- [ ] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildservers are happy. If not, fix **in this order**:
  - [ ] add a line in `doc/news/_preparation_next_release.md`
  - [ ] reformat the code with `scripts/dev/reformat-all`
  - [ ] make all unit tests pass
  - [ ] fix all memleaks
- [ ] The PR is rebased with current master.

If you have any troubles fulfilling these criteria, please write
about the trouble as comment in the PR. We will help you.
But we cannot accept PRs that do not fulfill the basics.

## Checklist

Check relevant points but **please do not remove entries**.
For docu fixes, spell checking, and similar none of these points below
need to be checked.

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://master.libelektra.org/doc/CODING.md))
- [ ] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned every code not directly written by me in [THIRD-PARTY-LICENSES](doc/THIRD-PARTY-LICENSES)

## Review

Reviewers will usually check the following:

- [ ] Documentation is introductory, concise, good to read and describes everything what the PR does
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

### CODE REVIEW

File: doc/news/_preparation_next_release.md [hunk 1]
```diff
@@ -82,6 +82,11 @@ The following section lists news about the [modules](https://www.libelektra.org/
 
 - We fixed an [use after free bug in the plugin](https://issues.libelektra.org/3561). _(René Schwaiger)_
 
+### GOpts
+
+- The `gopts` plugin can now be configured to use custom `argv` and `envp` values. _(Klemens Böswirth)_
+  WARNING: This feature is experimental and may be removed, replaced or otherwise changed in a future release.
```
Comment (markus2330) [inline doc/news/_preparation_next_release.md:88]: This is not necessary to say. Within `0.9.*` we can change anything anyway but not afterwards (depending on the status of the plugin).

What would be relevant to say if we actually plan to remove it and I hope we do as I do not like a global symbol for a setup. Ideally it should be in kdbOpen.

File: doc/news/_preparation_next_release.md [hunk 2]
```diff
@@ -82,6 +82,11 @@ The following section lists news about the [modules](https://www.libelektra.org/
 
 - We fixed an [use after free bug in the plugin](https://issues.libelektra.org/3561). _(René Schwaiger)_
 
+### GOpts
+
+- The `gopts` plugin can now be configured to use custom `argv` and `envp` values. _(Klemens Böswirth)_
```
Comment (markus2330) [inline doc/news/_preparation_next_release.md:87]: Please link to the files so that people see how to use that feature.

**Review by markus2330 — COMMENTED** [review]
Thank you for creating this soo quickly!

**Comment by markus2330** [comment]
@dev2718 is this PR in a way that you can use it?

**Comment by kodebach** [comment]
@markus2330 This PR is not intended to be merged in this state (that's why it's a draft). See also https://github.com/ElektraInitiative/libelektra/issues/3607#issuecomment-750864198

**Comment by markus2330** [comment]
As we want the FUSE implementation also in our master branch, we need to merge it at some point. I do not see a big problem in it, it is just one out of many cleanup task for the kdbOpen integration #2764, which is a must-have before 1.0 anyway.

**Comment by kodebach** [comment]
If you want to, we can of course merge this PR. But I'm almost certain I'll remove `elektraGOptsSetup` (and the C++ version) for the integration into `kdbOpen`. That's why I added the warnings and didn't really add easy-to-find documentation. In the mid-to-long term, nobody's really supposed to use this function.

**Comment by markus2330** [comment]
Is anything missing for this PR? Can we merge this for 0.9.4? Or do you plan to do the kdbOpen integration soon?

**Comment by kodebach** [comment]
If we have anything that depends on this being in a release, then we can merge it. However, if just need this in `master` I would merge it after the release. I don't know when we plan to release `0.9.5`, but I'll probably be done with integration of `kdbEnsure` into `kdbOpen` by then. I think I'll work on `kdbOpen` once #3606 is done. I'd estimate that the new `kdbOpen` will be done by the end of February.

**Comment by markus2330** [comment]
Ok, perfect. We'll merge it directly after the release. Ideally we will never release a version with elektraGOptsSetup.

**Comment by kodebach** [comment]
Superseeded by #3651

**Pull request closed without merging**